nequip/train/schedulefree.py

```diff
@@ -0,0 +1,64 @@
+from .lightning import NequIPLightningModule
+from typing import Dict, Any
+from lightning.pytorch.utilities.exceptions import MisconfigurationException
+from nequip.utils import RankedLogger
+
+logger = RankedLogger(__name__, rank_zero_only=True)
+
+
+class ScheduleFreeLightningModule(NequIPLightningModule):
+    """
+    NequIP LightningModule using Facebook's Schedule-Free optimizer.
+
+    This module wraps the model's optimizer in one of Facebook's Schedule-Free variants.
+    See: https://github.com/facebookresearch/schedule_free
+
+    Args:
+        optimizer (Dict[str, Any]): Dictionary that must include a `_target_`
+            corresponding to one of the Schedule-Free optimizers and other keyword arguments
+            compatible with the Schedule-Free variants.
+    """
+
+    def __init__(self, optimizer: Dict[str, Any], **kwargs):
+        valid_targets = {
+            "AdamWScheduleFree",
+            "SGDScheduleFree",
+            "RAdamScheduleFree",
+        }
+        if "_target_" not in optimizer or not any(
+            optimizer["_target_"].endswith(name) for name in valid_targets
+        ):
+            raise MisconfigurationException(
+                f"Invalid optimizer: expected Schedule-Free optimizer (_target_ ending with one of {valid_targets}), "
```

Comment:
nitpick, but looking at the CI error. We can also print "... but found {optimizer["_target_"]}" in the error message